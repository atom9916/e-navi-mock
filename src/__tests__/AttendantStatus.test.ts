/**
 * @jest-environment jsdom
 */

import { render, screen, fireEvent } from '@testing-library/vue'
import AttendantStatusVue from '@/components/AttendantStatus.vue'

describe('AttendantStatusコンポーネントのテスト', () => {
  test('コンポーネントが正しく表示される', () => {
    render(AttendantStatusVue)
    const label = screen.getByLabelText('出欠:')
    const input = screen.getByRole('textbox')

    expect(label).toBeInTheDocument()
    expect(input).toBeInTheDocument()
  })
  test('input要素をクリックするとドロップダウンリストが表示される', async () => {
    const attendantStatuses = ['出勤', '有給', '半休', '慶弔休', '欠勤', '休日出勤']

    render(AttendantStatusVue)
    const input = screen.getByRole('textbox')
    await fireEvent.click(input)

    const ul = screen.getByRole('list')
    const li = screen.getAllByRole('listitem')

    expect(ul).toBeInTheDocument()
    li.map((list, index) => {
      expect(list).toHaveTextContent(attendantStatuses[index])
    })
  })
  test('スナップショットテスト', () => {
    const { container } = render(AttendantStatusVue)
    expect(container).toMatchSnapshot()
  })
})
