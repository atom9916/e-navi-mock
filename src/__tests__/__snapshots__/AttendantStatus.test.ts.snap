// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttendantStatusコンポーネントのテスト スナップショットテスト 1`] = `
<div>
  
  <label
    for="attendance"
  >
    出欠:
  </label>
  <div
    class="dropdown"
  >
    <input
      id="attendance"
      type="text"
    />
    <ul
      class="dropdown-menu"
      style="display: none;"
    >
      
      <li>
        出勤
      </li>
      <li>
        有給
      </li>
      <li>
        半休
      </li>
      <li>
        慶弔休
      </li>
      <li>
        欠勤
      </li>
      <li>
        休日出勤
      </li>
      
    </ul>
  </div>
  
</div>
`;
